code/modules/mob/living/carbon/xenomorph/castes/widow/abilities_widow.dm

```diff
@@ -0,0 +1,266 @@
+	var/mob/living/carbon/xenomorph/X = owner
+
+	var/datum/ammo/xeno/acid/web/web_spit = GLOB.ammo_list[/datum/ammo/xeno/acid/web]
+
+	var/obj/projectile/newspit = new /obj/projectile(get_turf(X))
+
+	newspit.generate_bullet(web_spit, web_spit.damage * SPIT_UPGRADE_BONUS(X))
+	newspit.permutated += X
+	newspit.def_zone = X.get_limbzone_target()
+
+	newspit.fire_at(target, X, null, newspit.ammo.max_range)
+	succeed_activate()
+	add_cooldown()
+
+// ***************************************
+// *********** Burrow
+// ***************************************
+
+/datum/action/xeno_action/burrow
+	name = "Burrow"
+	ability_name = "Burrow"
+	mechanics_text = " Burrow into the ground to hide in plain sight "
+	action_icon_state = "savage_on" //temporary until I get my own icons
+	plasma_cost = 1
+	cooldown_timer = 1 SECONDS
+	keybind_signal = COMSIG_XENOABILITY_BURROW
+	/// This is to prevent crashing by registering the same signal twice
+	var/burrowed = FALSE
+
+/datum/action/xeno_action/burrow/action_activate()
+	. = ..()
+	if(burrowed)
+		return
+
+	var/mob/living/carbon/xenomorph/X = owner
+
+	to_chat(X, span_xenowarning("We start burrowing into the ground"))
+	if(!do_after(X, 1 SECONDS, TRUE, target, BUSY_ICON_DANGER))
+		return fail_activate()
+	to_chat(X, span_xenowarning("We have burrowed ourselves, we are hidden from the enemy"))
+
+	X.alpha = 0
+	X.mouse_opacity = 0
+	X.density = FALSE
+	X.throwpass = TRUE
+	burrowed = TRUE
+
+	for(var/mob/living/carbon/xenomorph/spiderling/kids in view(1, owner.loc))
+		RegisterSignal(kids, COMSIG_MOVABLE_MOVED, .proc/unburrow)
+		kids.alpha = 0
+		kids.mouse_opacity = 0
+		kids.density = FALSE
+		kids.throwpass = TRUE
+
+	RegisterSignal(X, COMSIG_MOVABLE_MOVED, .proc/unburrow)
+	succeed_activate()
+	add_cooldown()
+
+/datum/action/xeno_action/burrow/proc/unburrow(mob/M)
+	SIGNAL_HANDLER
+	if(!burrowed)
+		return
+	var/mob/living/carbon/xenomorph/X = owner
+	X.alpha = 255
+	X.mouse_opacity = 255
+	X.density = TRUE
+	X.throwpass = FALSE
+	burrowed = FALSE
+	UnregisterSignal(X, COMSIG_MOVABLE_MOVED)
+	succeed_activate()
+	add_cooldown()
+
+// ***************************************
+// *********** Leash Ball
+// ***************************************
+
+/datum/action/xeno_action/activable/leash_ball
+	name = "leash_ball" // change to in character name later
+	ability_name = "leash_ball" // change to in character name later
+	mechanics_text = " Spit a huge ball of web that snares groups of marines "
+	action_icon_state = "scatter_spit" // temporary until I get my own icons
+	plasma_cost = 1
+	cooldown_timer = 1 SECONDS
+	keybind_signal = COMSIG_XENOABILITY_LEASH_BALL
+
+/datum/action/xeno_action/activable/leash_ball/use_ability(atom/A)
+	var/turf/target = get_turf(A)
+	var/mob/living/carbon/xenomorph/X = owner
+	X.face_atom(target)
+	if(!do_after(X, 1 SECONDS, TRUE, X, BUSY_ICON_DANGER)) // currently here for balance prediction, shooting a 5x5 AoE snare is pretty insane even for T3 imo
```

Comment:
comments like that should not stick into repo. 
If you want to communicate with maintainers you should use the review button (so on github itself)